src/sentry/static/sentry/app/views/organizationDiscover/index.jsx

```diff
@@ -69,24 +68,16 @@ const OrganizationDiscover = createReactClass({
   },
 
   render: function() {
+    const {queryBuilder} = this.state;
```

Comment:
They way you're using `queryBuilder` is kind of odd here, esp having to call `forceUpdate`.

I think having QueryBuilder as a component calling a render prop makes a lot of sense here.

```javascript
<QueryBuilder project={project>
  {({internalQuery, externalQuery, actions: {updateField}}) => <OrganizationDiscovery {...props} onChange={updateField}/>}
</QueryBuilder>
```

And then tests would be just passing `<QueryBuilder>` a `jest.fn()` as a mock and checking what it was called with.

Reply:
I'll tackle that as a follow up. Wanted specifically to try this approach first here since making something that could be a pure function a component for the sake of rerendering also seems slightly non-ideal to me.